pageserver/src/tenant/timeline.rs

```diff
@@ -2852,9 +2701,7 @@ impl Timeline {
                 InMemoryLayer::create(self.conf, self.timeline_id, self.tenant_id, start_lsn)?;
             let layer_rc = Arc::new(new_layer);
 
-            layers.open_layer = Some(Arc::clone(&layer_rc));
-            layers.next_open_layer_at = None;
-
+            guard.open_new_layer(Arc::clone(&layer_rc));
```

Comment:
I feel there are two things happening in `get_layer_for_write`: first get layer, if it does not exist, open a new layer. Therefore, I feel it would better to simply move everything in the else branch into the layer manager as the rest of the logic are intended for "reads" while the actual "open" action should be done by the manager.

Reply:
https://github.com/neondatabase/neon/pull/4618/commits/7dc76a932a16c4d29932c145abf63c9cb354b2fe